Drafter upgrade & Tests: Integration of body attributes

honzajavorek opened the pull request:
Adding some integration/regression tests to the suite so Dredd is less prone to breakages introduced by [Drafter](https://github.com/apiaryio/drafter). Following picture is an illustration of how broken Dredd is at this moment:

<img width="613" alt="screen shot 2015-12-30 at 16 06 45" src="https://cloud.githubusercontent.com/assets/283441/12052441/7a028314-af0f-11e5-9447-47dd172976b5.png">

-----------------

**UPDATE 2016-01-04:** Upgraded to the latest Drafter version released today. Adjusted the tests a bit and now everything seems to be passing!

<img width="600" alt="screen shot 2016-01-04 at 18 26 10" src="https://cloud.githubusercontent.com/assets/283441/12095670/aa370e42-b310-11e5-98a4-150a05dcec24.png">

honzajavorek commented:
If merged, this should now fix https://github.com/apiaryio/dredd/issues/319 and https://github.com/apiaryio/protagonist/issues/113.

netmilk commented on test/fixtures/regression-319.apib:
```diff
@@ -0,0 +1,60 @@
+
++ id
++ name (string)
++ shoeSize: 42 (number)
```
I would definitely prefer some nested object example to be tested here. May we @honzajavorek?

honzajavorek replied:
Addressed in fdfd577.

netmilk commented on test/regressions/regression-319-test.coffee:
```diff
@@ -0,0 +1,185 @@
+  return results
+
+
+describe('Regression: Issue #319', ->
```
I'm sorry, I'm not sure about the style of bracket usage. Closing / parameter encapsulating brackets for functions aren't used in the entire project. Would you mind refactoring it?

honzajavorek replied:
I'm sorry, I got heavily used to following our [CoffeeScript Style Guide](https://github.com/apiaryio/coffeescript-style-guide) and I didn't realize I should check for an existing style in Dredd. Won't force parentheses here if it doesn't make sense in the context of the existing codebase.

honzajavorek replied:
Addressed in 786ad36.

netmilk commented on test/regressions/regression-319-test.coffee:
```diff
@@ -0,0 +1,185 @@
+    )
+  )
+
+  it('outputs no failures', ->
```
I would love to see an expectation for Dredd failing scenario when the backend doesn't reply with the expected response (missing required response key or different value type)

honzajavorek replied:
Addressed in a162380.

netmilk commented on test/fixtures/regression-319.apib:
```diff
@@ -0,0 +1,60 @@
+
+## Data Structures
+
+### User (object)
```
It would be awesome to have tested the `nullable` type attribute to be absolutely sure we're actually delivering solution for the #283

honzajavorek replied:
Addressed in fdfd577.

netmilk commented:
@honzajavorek This is perfect! I'm so excited! 💁🙅🙆🙋

Please feel to merge when my comments are addressed.

honzajavorek commented:
@netmilk Addressed comments, added negative scenarios. Everything seems to work. I'm a bit hesitant to do the merge myself though as I made significant changes.

w-vi commented:
@honzajavorek still looks good, so :+1:

netmilk commented:
Awesome! Merging!
